oozie-to-airflow/mappers/pig_mapper.py

```diff
@@ -64,16 +70,36 @@ def _parse_config(self):
                     )
                     self.properties[name] = value
 
-    def convert_to_text(self):
+    def convert_to_text(self) -> str:
         template_loader = jinja2.FileSystemLoader(searchpath=os.path.join(ROOT_DIR, "templates/"))
         template_env = jinja2.Environment(loader=template_loader)
         template = template_env.get_template(self.template)
         prepare_command = self.get_prepare_command(self.oozie_node, self.params)
         return template.render(prepare_command=prepare_command, **self.__dict__)
 
+    def copy_extra_assets(self, input_directory_path: str, output_directory_path: str):
```

Comment:
This method is quite long and there are a few levels of indentation. How about splitting it up into 2-3 methods to increase readability?

My suggestion (a bit rough, would need polishing):
```
def copy_extra_assets(self, input_directory_path: str, output_directory_path: str):
    _check_paths(input_directory_path, output_directory_path)
    source_pig_file_path = os.path.join(input_directory_path, self.script_file_name)
    destination_pig_file_path = os.path.join(output_directory_path, self.script_file_name)
    _overwrite_script(destination_pig_file_path, self, source_pig_file_path)


def _overwrite_script(destination_pig_file_path, self, source_pig_file_path):
    with open(destination_pig_file_path, "w") as destination_pig_file:
        with open(source_pig_file_path, "r") as source_pig_file:
            pig_script = source_pig_file.read()
            if self.files or self.archives:
                _add_symlinks(destination_pig_file, self)
            destination_pig_file.write(pig_script)


def _add_symlinks(destination_pig_file, self):
    destination_pig_file.write("set mapred.create.symlink yes;\n")
    if self.files:
        destination_pig_file.write("set mapred.cache.file {};\n".format(self.hdfs_files))
    if self.archives:
        destination_pig_file.write("set mapred.cache.archives {};\n".format(self.hdfs_archives))


def _check_paths(input_directory_path, output_directory_path):
    if not input_directory_path:
        raise Exception("The input_directory_path should be set and is {}".format(input_directory_path))
    if not output_directory_path:
        raise Exception("The output_directory_path should be set and is {}".format(output_directory_path))
```

Reply:
👍